### PR TITLE
feat: support sample_age_limit for QueueConfig

### PR DESCRIPTION
**Pull request by dongjiang1989**
## Description

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue._

Support new option in QueueConfig: "sample_age_limit".
ref: https://github.com/prometheus/prometheus/pull/13002

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [x] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Verification

Please check the [Prometheus-Operator testing guidelines](../TESTING.md) for recommendations about automated tests.

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
support new option in QueueConfig: "sample_age_limit".
```

### CODE REVIEW

File: pkg/apis/monitoring/v1/prometheus_types.go [hunk 1]
```diff
@@ -1313,6 +1313,8 @@ type QueueConfig struct {
 	// Retry upon receiving a 429 status code from the remote-write storage.
 	// This is experimental feature and might change in the future.
 	RetryOnRateLimit bool `json:"retryOnRateLimit,omitempty"`
+	// SampleAgeLimit is samples older than the limit will be dropped.
+	SampleAgeLimit string `json:"sampleAgeLimit,omitempty"`
```
Comment (simonpasquier) [inline pkg/apis/monitoring/v1/prometheus_types.go:1317]:
```suggestion
	// It requires Prometheus >= v2.50.0.
	//
	// +optional
	SampleAgeLimit *Duration `json:"sampleAgeLimit,omitempty"`
```

Comment (dongjiang1989) [reply pkg/apis/monitoring/v1/prometheus_types.go:1317]: Thanks. Got it.

File: pkg/apis/monitoring/v1/prometheus_types.go [hunk 2]
```diff
@@ -1313,6 +1313,11 @@ type QueueConfig struct {
 	// Retry upon receiving a 429 status code from the remote-write storage.
 	// This is experimental feature and might change in the future.
 	RetryOnRateLimit bool `json:"retryOnRateLimit,omitempty"`
+	// SampleAgeLimit is samples older than the limit will be dropped.
+	// It requires Prometheus >= v2.50.0.
+	//
+	// +optional
+	SampleAgeLimit string `json:"sampleAgeLimit,omitempty"`
```
Comment (simonpasquier) [inline pkg/apis/monitoring/v1/prometheus_types.go:1320]: there's a built-in type for validating durations.
```suggestion
	SampleAgeLimit *Duration `json:"sampleAgeLimit,omitempty"`
```

Comment (dongjiang1989) [reply pkg/apis/monitoring/v1/prometheus_types.go]: Thanks.  Change string to Duration done.

**Comment by dongjiang1989** [comment]
@simonpasquier PTAL, thanks

**Review by simonpasquier — COMMENTED** [review]
Thanks!

**Comment by simonpasquier** [comment]
You'd need to resolve conflicts though.

**Comment by dongjiang1989** [comment]
> You'd need to resolve conflicts though.

Thanks. Fixed it

File: pkg/apis/monitoring/v1/prometheus_types.go [hunk 3]
```diff
@@ -1322,6 +1322,11 @@ type QueueConfig struct {
 	// Retry upon receiving a 429 status code from the remote-write storage.
 	// This is experimental feature and might change in the future.
 	RetryOnRateLimit bool `json:"retryOnRateLimit,omitempty"`
+	// SampleAgeLimit is samples older than the limit will be dropped.
```
Comment (ArthurSens) [inline pkg/apis/monitoring/v1/prometheus_types.go:1325]:
```suggestion
	// SampleAgeLimit drops samples older than the limit.
```

**Review by ArthurSens — APPROVED** [review]
LGTM, just one wording nit

File: pkg/apis/monitoring/v1/prometheus_types.go (hunk 3 above)
Comment (dongjiang1989) [reply pkg/apis/monitoring/v1/prometheus_types.go]: Thanks. Update it.

**Pull request merged**